pya/arecorder.py

```diff
@@ -39,8 +39,11 @@ def __init__(self, sr=44100, bs=256, device=None, channels=None, backend=None, *
         self._record_all = True
         self.gains = np.ones(self.channels)
         self.tracks = slice(None)
-        self.device = device or self.backend.get_default_input_device_info()['index']
-        self.channels = channels or self.backend.get_default_input_device_info()['maxInputChannels']
+        self._device = device or self.backend.get_default_input_device_info()['index']
+        if channels:
```

Comment:
I suggest to simplify as 
self.channels = channels or
     self.backend.get_device_info_by_index(self._device)['maxInputChannels']
(similiarly how it is done with device above....)

Reply:
Ok. I was thinking of not to have overly long line. But it is more consistent with others I guess